lib/feedsme/index.js

```diff
@@ -54,153 +64,223 @@ function pkgName(pkg) {
+  // which get promoted that were established upon publish
+  //
+  const releaseVersion = releaseLine.dependents && releaseLine.dependents[name];
+  const publish = env === 'dev';
 
-        //
-        // Remove all the datastar model garbage so we can get a clean object to
-        // manipulate and pass along.
-        //
-        pkg = pkg.toJSON();
-        app.log.debug(`Finding version information of ${ pkgName(pkg) }`);
-
-        async.parallel({
-          version: Version.get.bind(Version, pkg.name + '@' + pkg.version),
-          head: BuildHead.get.bind(BuildHead, { env, name: pkg.name })
-        }, (e, { version, head = { version: '0.0.0' } }) => {
-          if (!version || e) {
-            if (e) {
-              app.log.debug(`Failed to lookup dependent ${ pkg } of ${ name } `, err);
-            } else {
-              app.log.debug(`Unable to find ${ pkgName(pkg) }`);
-            }
-
-            return fn(err);
-          }
-
-          app.log.info(`Getting attachments for ${ pkg.name }`);
-          version.getAttachment(npm, function attached(errored, body) {
-            if (!body || errored) {
-              if (errored) {
-                app.log.debug(`Failed to lookup attachment for ${ pkgName(pkg) }, received error: ${ errored.message }`);
-              } else {
-                app.log.debug(`No attachment available for ${ pkgName(pkg) }`);
-              }
-
-              return fn(err);
-            }
-
-            data = JSON.parse(body.value);
-            data._attachments = body._attachments;
-
-            Object.keys(pkg).forEach(function merge(key) {
-              if (key in data) return;
-              //
-              // Specially handle dist-tags to make sure we have the latest
-              // for the particular package since this is stored as camelCase
-              // since we cannot store dash based keys in C*
-              //
-              if (key === 'distTags') {
-                data['dist-tags'] = pkg[key];
-                return;
-              }
-
-              data[key] = pkg[key];
-            });
-
-            //
-            // Force the environment on which we need to build in as this is
-            // what we received as initiator.
-            //
-            data.env = env;
-            //
-            // Update the version so we arent overwriting the current latest
-            // version when triggering a dependent build
-            //
-            const { payload, action } = feedsme.increment(data, head);
-            app.log.info(`Triggering ${action} action for ${ data.name }`);
-            feedsme[action](payload, fn);
-          });
-        });
-      });
-    }, done);
+  let pkg = await Package.get(name);
+  if (!pkg) return this.log.debug(`Package ${ name } not found in cassandra`);
+
+  pkg = pkg.toJSON();
+
+  //
+  // If our newest version of our rootPkg published here is a new major
```

Comment:
similar to my comment before, should this actually look at the semver range decided by the dependent package on the dependee?  We're assuming it's effectively a `^1.0.0` style  range.  Thats probably okay for simplification now, but should probably be documented if we do it that way.